### PR TITLE
Add tests of body-bound top-level collections

### PR DESCRIPTION
**Pull request by dougbu**
- relates to #4496

Also fix test names in `BodyValidationIntegrationTests`

### CODE REVIEW

**Comment by javiercn** [comment]
Looks good to me :shipit:

File: test/Microsoft.AspNetCore.Mvc.Formatters.Json.Test/JsonInputFormatterTest.cs [hunk 1]
```diff
@@ -138,6 +138,70 @@ public static IEnumerable<object[]> JsonFormatterReadSimpleTypesData
         }
 
         [Fact]
+        public async Task ReadAsync_ReadsValidArray()
+        {
+            // Arrange
+            var content = "[0, 23, 300]";
+            var logger = GetLogger();
+            var formatter =
+                new JsonInputFormatter(logger);
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Formatters.Json.Test/JsonInputFormatterTest.cs]: nitpick: linebreak isn't needed here

File: test/Microsoft.AspNetCore.Mvc.IntegrationTests/BodyValidationIntegrationTests.cs [hunk 2]
```diff
@@ -28,7 +28,7 @@ private class Address
         }
 
         [Fact]
-        public async Task ModelMetaDataTypeAttribute_ValidBaseClass_NoModelStateErrors()
+        public async Task ModelMetadataTypeAttribute_ValidBaseClass_NoModelStateErrors()
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.IntegrationTests/BodyValidationIntegrationTests.cs]: :heart_eyes_cat:

**Comment by rynowak** [comment]
:shipit:

File: test/Microsoft.AspNetCore.Mvc.Formatters.Json.Test/JsonInputFormatterTest.cs (hunk 1 above)
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Formatters.Json.Test/JsonInputFormatterTest.cs]: Fixed when I pushed to origin.

**Comment by dougbu** [comment]
d9141d6

**Pull request closed without merging**